src/system.jl

```diff
@@ -25,6 +25,47 @@ end
 const Zones = Dictionary{ZoneNum, Zone}
 
 const UnitCode = Int64
+const GenId = Union{UnitCode, InlineString31}
+
+"""
+    $TYPEDEF
+
+Abstract type to describe generation-specific internals of generators.
+"""
+abstract type Technology end
+
+"""
+    $TYPEDEF
+
+Type to describe a single configuration of a combined cycle generator.
+
+Fields:
+$TYPEDFIELDS
+"""
+Base.@kwdef struct CombinedCycle <: Technology
+    "Id linking back to the resource this configuration is from"
+    parent::GenId
+    "Symbol describing the technology of the generator"
+    fuel_type::Symbol
```

Comment:
Does this supply enough information? Do we need more fields?

Reply:
I think this is sufficient but we should get PS to confirm for sure.